apps/emqx_management/src/emqx_mgmt_api.erl

```diff
@@ -355,6 +394,8 @@ apply_total_query(QueryState = #{table := Tab}) ->
             Fun(Tab)
     end.
 
+counting_total_fun(_QueryState = #{qs := {[], []}, options := #{fast_total_counting := true}}) ->
+    fun(Tab) -> ets:info(Tab, size) end;
```

Comment:
Do we really need to introduce another option here? I'm a bit wary of the possibility that it may easily be conflicting with query's matchspec, if accidentally misused.

I'd propose that we allow `match_spec` to be `undefined` instead. Then, when we know that some query is equivalent to full table scan, we may add another clause, for example in `emqx_mgmt_api_clients`.
```
-spec qs2ms(list()) -> ets:match_spec() | undefined.
qs2ms([]) ->
    undefined;
qs2ms(Qs) ->
    {MtchHead, Conds} = qs2ms(Qs, 2, {#{}, []}),
    [{{'$1', MtchHead, '_'}, Conds, ['$_']}].
```

And then handle it in `emqx_mgmt_api`, i.e.:
```
            case maps:get(continuation, QueryState, undefined) of
                undefined when Ms == undefined ->
                    ets:select(Tab, {'_', [], ['$_']}, Limit);
                undefined ->
                    ets:select(Tab, Ms, Limit);
```

We would still need to go over the code and mark the full table scans with `match_spec => undefined` but at least this couldn't be inconsistent. What do you think?

Reply:
Thanks for your review.  I got your concern but the implementation may introduce another inconsistency possibility🤔  : **"The caller must specify to return the entire record structure when constructing the MatchSpec"**. i.e:
```
qs2ms([]) ->
    undefined;
qs2ms(Qs) ->
     %% Only returns the first element for the record
     %% BUT it difference with '$_'  of the `Ms == undefined` case clause
      [{{'$1', MtchHead, '_'}, Conds, ['$1']}]
```
But my concern may be a bit redundant, due to all callers now returning the entire record for now🥲 .

Reply:
You're right, indeed it isn't suitable for such scenarios. As I see it, we'd have a few choices then:
* bring that same option (which introduces same inconsistency, though we could hide behind some DSL to make it a bit harder to misuse),
* disallow that (which weakens flexibility),
* perform syntactical analysis of the matchspec (which sounds way too complex).

But anyway, that was just a suggestion, it's fine to go either way.

Reply:
Understood. Let's keep the current implementation for now. It is simple enough. and if we encounter any new issues in the future, we can fix them on a case-by-case.😉